router/src/server.rs

```diff
@@ -1698,33 +1696,35 @@ pub async fn run(
 
     // Only send usage stats when TGI is run in container and the function returns Some
     let is_container = matches!(usage_stats::is_container(), Ok(true));
-
-    let user_agent = if !disable_usage_stats && is_container {
-        let reduced_args = usage_stats::Args::new(
-            config.clone(),
-            tokenizer_config.tokenizer_class.clone(),
-            max_concurrent_requests,
-            max_best_of,
-            max_stop_sequences,
-            max_top_n_tokens,
-            max_input_tokens,
-            max_total_tokens,
-            // waiting_served_ratio,
-            // max_batch_prefill_tokens,
-            // max_batch_total_tokens,
-            // max_waiting_tokens,
-            // max_batch_size,
-            revision.clone(),
-            validation_workers,
-            messages_api_enabled,
-            disable_grammar_support,
-            max_client_batch_size,
-            disable_usage_stats,
-            disable_crash_reports,
-        );
-        Some(usage_stats::UserAgent::new(reduced_args))
-    } else {
-        None
+    let user_agent = match usage_stats_level {
+        usage_stats::UsageStatsLevel::On | usage_stats::UsageStatsLevel::NoStack
+            if is_container =>
+        {
```

Comment:
```suggestion
    let user_agent = match (usage_stats_level, is_container) {
        (usage_stats::UsageStatsLevel::On | usage_stats::UsageStatsLevel::NoStack, true)  =>
        {
        _ => None
```

Not necessarily better but I tend to try and limit the number of branches whenever possible, so codepaths are easier to reason about.

Reply:
yeah, this makes sense to me 👍 thank you!